### PR TITLE
add setting validate, update readme and .envExample

### PR DESCRIPTION
**Pull request by VictoriqueMoe**
adds a settings validation that will make sure you have the correct `.env` settings needed to run the application

### CODE REVIEW

File: .envExample [hunk 1]
```diff
@@ -1,7 +1,11 @@
...
 HTTPS_PORT=4000
 SESSION_KEY=YourSessionKey
 NODE_ENV=development
-BASE_URL=http://localhost:8081
-FILE_SIZE_UPLOAD_LIMIT_MB=512
\ No newline at end of file
+BASE_URL=http://127.0.0.1:8089
+FILE_SIZE_UPLOAD_LIMIT_MB=512
+ALLOWED_FILES=wad,pk3,zip
+ALLOWED_HEADERS=PWAD,PK,PK
```
Comment (harristmv) [inline .envExample:9]: is `PK,PK` the same? (also it's not readable)

File: README.md [hunk 2]
```diff
@@ -15,35 +21,5 @@
     npm run build
     npm run start:prod
 ```
-    
-## Docker
```
Comment (harristmv) [inline README.md:19]: you don't need to delete this section. It works fine, also for `start:prod`.

Just update documentation: 
set SMTP variables in `.env` after running `npm run start` 

```batch
SMTP_HOST=smtp.ethereal.email
SMTP_PORT=587
SMTP_USER=<autogenerated>
SMTP_PASS=<autogenerated>
```

```batch
docker compose build
docker compose up
```

**Review by harristmv — COMMENTED** [review]
Making docker work is kinda hacky but it does the job - you just need to rerun `npm start` from outside. You could set these variables maybe when running `build`

File: .envExample (hunk 1 above)
Comment (VictoriqueMoe) [reply .envExample:9]: it's a mapping 
```
ALLOWED_FILES=wad,pk3,zip
ALLOWED_HEADERS=PWAD,PK,PK
```
`WAD` = `PWAD`
`pk3` = `PK`
`zip` = `PK`

zip and pk3 are the same

**Pull request merged**